### PR TITLE
Fix race condition in integration tests.

### PR DESCRIPTION
**Pull request by tcsc**
Some integration tests modify global "constants" to speed up test
execution (e.g. shortening polling intervals). This is occasionally
tripping the Go data race detector, so I have added explicit
serialisation to reading and writing these global settings.

These values are only ever changed in a test environment, and there
should be zero contention for them in a non-test environment.

### CODE REVIEW

File: api/defaults/defaults.go [hunk 1]
```diff
@@ -50,17 +51,39 @@ const (
 )
 
 var (
-	// ServerKeepAliveTTL is a period between server keep-alives,
+	moduleLock sync.RWMutex
+
+	// svrKeepAliveTTL is a period between server keep-alives,
```
Comment (codingllama) [inline api/defaults/defaults.go:56]:
```suggestion
	// serverKeepAliveTTL is a period between server keep-alives,
```

File: api/defaults/defaults.go [hunk 2]
```diff
@@ -50,17 +51,39 @@ const (
...
 	// when servers announce only presence without sending full data
-	ServerKeepAliveTTL = 60 * time.Second
+	serverKeepAliveTTL = 60 * time.Second
 
 	// KeepAliveInterval is interval at which Teleport will send keep-alive
```
Comment (codingllama) [inline api/defaults/defaults.go:60]:
```suggestion
	// keepAliveInterval is interval at which Teleport will send keep-alive
```

**Review by codingllama — COMMENTED** [review]
The real culprit here seems to be SetTestTimeouts, who decided to tweak constants instead of creating servers with an explicitly lower timeout (possible for some of the types we are using here, but not others).

The best scenario would be for the defaults to be truly `const` and for us to explicitly set lower values during server creation, but that might translate to a lot of work for what should be a somewhat simple change.

Is it be possible to change the var values _before_ any data races happen? Say, during test setup, before any servers/clients are created?

**Comment by nklaassen** [comment]
Yeah I've got to agree with @codingllama ideally we would not mess with global variables at all, but I would be okay with this change if the alternative is too much effort, we add a TODO comment, and file a bug to fix it

**Comment by tcsc** [comment]
> The real culprit here seems to be SetTestTimeouts, who decided to tweak constants instead of creating servers with an explicitly lower timeout (possible for some of the types we are using here, but not others).
> 
> The best scenario would be for the defaults to be truly `const` and for us to explicitly set lower values during server creation, but that might translate to a lot of work for what should be a somewhat simple change.

Totally agreed: In a perfect world _we should not be messing with these values in any way_. The _correct_ thing to do is get rid of the notion `SetTestTimeouts()`, surface the values it is tweaking in config and create the servers properly in the first place. Unfortunately, 

1. There is a _lot_ of surgery required for this, and 
2. This is just one in a series of data race fixes that are kind of orthogonal to the actual work I'm doing (i.e. moving our CI to GCB), and I want to get _that_ finished sometime this decade :-)

I'm happy to mark it with a `TODO` and a ticket for now, and circle back once I've finished the CI port, if that works for everyone.

**Review by codingllama — APPROVED** [review]

**Comment by codingllama** [comment]
> I'm happy to mark it with a TODO and a ticket for now, and circle back once I've finished the CI port, if that works for everyone.

Sure, sounds good. Thanks for the explanation.

**Review by nklaassen — APPROVED** [review]

**Comment by tcsc** [comment]
See #8913

**Review by codingllama — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Pull request merged**